GUI/src/main/kotlin/ui/components/FileSelectorButton.kt

```diff
@@ -65,3 +65,14 @@ fun RowScope.FileSelectorButton(
         }
     }
 }
+
+/**
+ * Opens a file chooser dialog and returns the selected file path.
+ *
+ * @return The selected file path, or null if no file was selected.
+ */
+fun openFileChooserAndGetPath(): String? {
+    val fileChooser = JFileChooser()
+    val result = fileChooser.showOpenDialog(null)
```

Comment:
I found out that on Mac you cannot Save to a non existing file, because I cannot name it. Please use `fileChooser.showSaveDialog(null)`.